src/aui/dockart.cpp

```diff
@@ -88,6 +89,22 @@ wxBitmap wxAuiBitmapFromBits(const unsigned char bits[], int w, int h,
     return wxBitmap(img);
 }
 
+// wxAuiScaleBitmap is a utility function that scales a TabArt bitmap
+wxBitmap wxAuiScaleBitmap(const wxBitmap& bmp, double scale)
+{
+#if wxUSE_IMAGE && !defined(__WXGTK3__) && !defined(__WXMAC__)
+    // scale to a close round number to improve quality
+    scale = floor(scale + 0.25);
+    if (scale > 1.0 && !(bmp.GetScaleFactor() > 1.0))
```

Comment:
I'm still uncomfortable with checking bitmap scale factor as the indication of whether it was scaled or not. We should store the scale factor the bitmap is good for (`1.0` initially) instead.

Reply:
I'm not quite sure how this needs to be done (or how this scale factor would be modified). Are you proposing adding a new API call? If you, can you take a stab at this and I'll test the changes?